CMakeLists.txt

```diff
@@ -14,6 +16,50 @@ if (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
   endif(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
 endif (NOT DEFINED LLVM_SPIRV_BUILD_EXTERNAL)
 
+# Download spirv.hpp from the official SPIRV-Headers repository.
+# One can skip this step by manually setting
+# LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR path.
+if(NOT DEFINED LLVM_TOOL_SPIRV_HEADERS_BUILD AND
+   NOT DEFINED LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR)
+  set(LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR
+    "${CMAKE_CURRENT_BINARY_DIR}/SPIRV-Headers")
+  message(STATUS "SPIR-V Headers location is not specified. Will try to download
+          spirv.hpp from https://github.com/KhronosGroup/SPIRV-Headers into
+          ${LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR}")
+  ExternalProject_Add(spirv-headers
+    GIT_REPOSITORY    https://github.com/KhronosGroup/SPIRV-Headers.git
+    GIT_TAG           origin/master
```

Comment:
Relying on the `master` branch assumes that the translator will always build with the latest upstream headers.  I'm not sure if that's always the case?  I wonder if we should instead rely on a known good revision of the spirv-headers repo, and maintain a SHA-1 somewhere (similar to what [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools/blob/a95bc460f924a9541dc3a85745dcb15140c8fcf3/DEPS#L9) does with its `DEPS` file)?

Reply:
That's a good point, we could switch to using a hash. This should also include a mention in the docs for implementing extensions, i.e. a note about updating the hash after the developer's patch has been accepted into SPIR-V Headers. In a way, this could solve the maintenance problem, as the newer hashes of the SPIR-V Headers will have be tested with the translator sources by the extension developers, who could solve the unexpected build failures alongside their implementations.

Reply:
Agree

Reply:
We will need to have a single point where we maintain that hash, because both CMakeLists.txt and github CI need it.  Unfortunately I do not have a good suggestion how to do so.

Reply:
Fair point; I'll try and come up with something.

Reply:
Done; switched to using a dedicated `.conf` file to read from both CMake and YML scrupts.